magicblock-committor-service/src/persist/commit_persister.rs

```diff
@@ -1,258 +1,574 @@
+    //     self.db.lock().expect(POISONED_MUTEX_MSG).
+    // }
+}
+
+/// Blanket implementation for Option
+impl<T: IntentPersister> IntentPersister for Option<T> {
+    fn start_base_intents(
+        &self,
+        l1_messages: &[ScheduledBaseIntent],
+    ) -> CommitPersistResult<()> {
+        match self {
+            Some(persister) => persister.start_base_intents(l1_messages),
+            None => Ok(()),
+        }
+    }
+
+    fn start_base_intent(
+        &self,
+        l1_message: &ScheduledBaseIntent,
+    ) -> CommitPersistResult<()> {
+        match self {
+            Some(persister) => persister.start_base_intent(l1_message),
+            None => Ok(()),
+        }
+    }
+
+    fn set_commit_id(
         &self,
-        reqid: &str,
+        message_id: u64,
+        pubkey: &Pubkey,
+        commit_id: u64,
+    ) -> CommitPersistResult<()> {
+        match self {
+            Some(persister) => {
+                persister.set_commit_id(message_id, pubkey, commit_id)
+            }
+            None => Ok(()),
+        }
+    }
+
+    fn set_commit_strategy(
+        &self,
+        commit_id: u64,
+        pubkey: &Pubkey,
+        value: CommitStrategy,
+    ) -> CommitPersistResult<()> {
+        match self {
+            Some(persister) => {
+                persister.set_commit_strategy(commit_id, pubkey, value)
+            }
+            None => Ok(()),
+        }
+    }
+
+    fn update_status_by_message(
+        &self,
+        message_id: u64,
+        pubkey: &Pubkey,
+        status: CommitStatus,
+    ) -> CommitPersistResult<()> {
+        match self {
+            Some(persister) => {
+                persister.update_status_by_message(message_id, pubkey, status)
+            }
+            None => Ok(()),
+        }
+    }
+
+    fn update_status_by_commit(
+        &self,
+        commit_id: u64,
+        pubkey: &Pubkey,
+        status: CommitStatus,
+    ) -> CommitPersistResult<()> {
+        match self {
+            Some(persister) => {
+                persister.update_status_by_commit(commit_id, pubkey, status)
+            }
+            None => Ok(()),
+        }
+    }
+
+    fn get_commit_statuses_by_message(
+        &self,
+        message_id: u64,
     ) -> CommitPersistResult<Vec<CommitStatusRow>> {
-        self.db.get_commit_statuses_by_reqid(reqid)
+        match self {
+            Some(persister) => {
+                persister.get_commit_statuses_by_message(message_id)
+            }
+            None => Ok(Vec::new()),
+        }
     }
 
-    pub fn get_commit_status(
+    fn get_commit_status_by_message(
         &self,
-        reqid: &str,
+        message_id: u64,
         pubkey: &Pubkey,
     ) -> CommitPersistResult<Option<CommitStatusRow>> {
-        self.db.get_commit_status(reqid, pubkey)
+        match self {
+            Some(persister) => {
+                persister.get_commit_status_by_message(message_id, pubkey)
```

Comment:
Here we're mixing and matching two things into one. This is what this meant before:

- bundle_id: id of a bundle of accounts that have to be committed together (Atomicity) which has to be honored on retries as well
- req_id: the id of the specific request to commit something which is useful to give user feedback about processing status of a specific commit
The idea was that the req_id could update for retries.
If that is not the case in the way you are planning to implement retries then I suppose it is fine to merge them.

Reply:
There's no bundles just messages/intents that executed one by one and never bundled as per protocol.
req_id = message_id/intent_id